### PR TITLE
More simplified Pulumi based deployment walkthrough

### PR DESCRIPTION
**Pull request by viveklak**
### Proposed changes

Added Pulumi programs to perform the Operator install instead of asking users to transcribe code. I feel it simplifies the experience quite a bit. Also ran into https://github.com/pulumi/pulumi-kubernetes/issues/1481 which I added a workaround for.

### CODE REVIEW

**Comment by viveklak** [comment]
@lblackstone @jaxxstorm - took a slightly different tack to document the operator deployment instructions. In this case, the change is to embed full-fledged programs in the release tarball for installing the operator. The instructions now point the user to using the relevant Pulumi project instead of asking users to copy/paste code. Look forward to your thoughts.

File: deploy/deploy-operator-go/main.go [hunk 1]
```diff
@@ -0,0 +1,306 @@
...
+
+func main() {
+	pulumi.Run(func(ctx *pulumi.Context) error {
+		// Download file
+		filePath, cleanup, err := downloadFile("https://raw.githubusercontent.com/pulumi/pulumi-kubernetes-operator/v1.2.0/deploy/crds/pulumi.com_stacks.yaml")
+		if err != nil {
+			return err
+		}
+		crds, err := yaml.NewConfigFile(ctx, "crds", &yaml.ConfigFileArgs{
+			File: filePath,
```
Comment (lblackstone) [inline deploy/deploy-operator-go/main.go:28]: You can supply a URL directly to the File argument. Was that not working for some reason?

File: deploy/deploy-operator-go/main.go [hunk 2]
```diff
@@ -0,0 +1,306 @@
...
+		})
```
Comment (lblackstone) [inline deploy/deploy-operator-go/main.go:29]: Do you not need to remove the status field with a transformation here as well?

File: deploy/deploy-operator-cs/MyStack.cs [hunk 3]
```diff
@@ -0,0 +1,289 @@
...
+using Pulumi.Kubernetes.Types.Inputs.Apps.V1;
+using Pulumi.Kubernetes.Types.Inputs.Meta.V1;
+using Pulumi.Kubernetes.Types.Inputs.Rbac.V1;
+
+class MyStack : Stack
+{
+    public MyStack()
+    {
+        var crds = new Kubernetes.Yaml.ConfigFile("crds", new Kubernetes.Yaml.ConfigFileArgs{
+            File = "https://raw.githubusercontent.com/pulumi/pulumi-kubernetes-operator/v1.2.0/deploy/crds/pulumi.com_stacks.yaml"
```
Comment (lblackstone) [inline deploy/deploy-operator-cs/MyStack.cs:13]: Need to remove status field with transformation?

**Review by lblackstone — COMMENTED** [review]
I think the general approach looks good. Just a few questions about the implementation, and it would be good to make sure the programs all run as expected.

File: deploy/deploy-operator-go/main.go (hunk 1 above)
Comment (viveklak) [reply deploy/deploy-operator-go/main.go:28]: I forgot to put a link to it here it seems but I also ran into https://github.com/pulumi/pulumi-kubernetes/issues/1569. Looks. like remote config files are not handled in the Go implementation.

File: deploy/deploy-operator-go/main.go (hunk 2 above)
Comment (viveklak) [reply deploy/deploy-operator-go/main.go:29]: It seems to only "fail" on the python implementation. I can add the transformation but it didn't seem to have an impact with running the code to completion.

File: deploy/deploy-operator-cs/MyStack.cs (hunk 3 above)
Comment (viveklak) [reply deploy/deploy-operator-cs/MyStack.cs:13]: See above.

File: deploy/deploy-operator-go/main.go (hunk 1 above)
Comment (lblackstone) [reply deploy/deploy-operator-go/main.go:28]: Oh, hmm. Didn't realize that.

**Review by lblackstone — APPROVED** [review]

**Pull request merged**